synapse/models/base.py

```diff
@@ -0,0 +1,79 @@
+import logging
+
+import synapse.common as s_common
+
+import synapse.lib.tufo as s_tufo
+import synapse.lib.module as s_module
+
+logger = logging.getLogger(__name__)
+
+class BaseMod(s_module.CoreModule):
+
+    @staticmethod
+    def getBaseModels():
+        modl = {
+
+            'types': (
+
+                ('source', {
+                    'subof': 'guid',
+                    'doc': 'A unique source of data.'}),
+
+                ('seen', {
+                    'subof': 'comp',
+                    'fields': 'source=source,node=node:ndef',
+                    'doc': 'Annotates that the data in a node was obtained from or observed by a given source.'}),
+
+                ('record', {
+                    'subof': 'guid',
+                    'doc': 'A flexible clustering type used for custom records that reference other nodes.'}),
+
+                ('recref', {
+                    'subof': 'comp',
+                    'fields': 'record=record,node=ndef',
+                    'doc': 'A record reference used to connect records to nodes which they group'}),
+            ),
+
+            'forms': (
+
+                ('source', {}, (
+                    ('name', {'ptype': 'str:lwr',
+                        'doc': 'A human friendly name for the source.'}),
+                    ('type', {'ptype': 'str:lwr',
+                        'doc': 'An optional type field used to group sources.'}),
+                )),
+
+                ('seen', {}, (
```

Comment:
The docs here are a little bit confusing. Does this node represent that the source saw something or that this is when we saw something from the source. Is the time:min/time:max the time that we saw it or that they saw it?

Reply:
This is fine after discussion w/ @therealsilence